src/components/AllocationBar/AllocationBar.tsx

```diff
@@ -0,0 +1,151 @@
+  }, [values, onChange])
+
+  // Start drag
+  const onHandleMouseDown = (idx: number) => (e: React.MouseEvent) => {
+    setDragIndex(idx)
+    e.preventDefault()
+  }
+
+  // Drag logic
+  const onMouseMove = ({ clientX }: MouseEvent) => {
+    if (dragIndex === null || !barRef.current) return
+
+    const rect = barRef.current.getBoundingClientRect()
+    const x = clamp(clientX - rect.left, 0, rect.width)
+
+    // Calculate px positions
+    let cumSum = 0
+    for (let i = 0; i < dragIndex; ++i) cumSum += values[i]
+    const leftPx = (cumSum / 100) * rect.width
+    const rightPx = ((cumSum + values[dragIndex] + values[dragIndex + 1]) / 100) * rect.width
+
+    // Boundaries to keep segments from getting too small
+    const pairSum = values[dragIndex] + values[dragIndex + 1]
+
+    // Handler position as a proportion between minX and maxX
+    const totalPairPx = rightPx - leftPx
+    const minLeftPx = (MIN_SEGMENT_PERCENT / pairSum) * totalPairPx
+    const maxLeftPx = totalPairPx - minLeftPx
+
+    // Calculate the handle's x relative to leftPx
+    let relX = clamp(x - leftPx, minLeftPx, maxLeftPx)
+
+    // Convert back to percentage and round to whole numbers
+    let leftValue = Math.round((relX / totalPairPx) * pairSum)
+    let rightValue = pairSum - leftValue
+
+    // When a segment reaches minimum size, maintain the right segment's width
+    if (leftValue < MIN_SEGMENT_PERCENT) {
+      leftValue = Math.ceil(MIN_SEGMENT_PERCENT)
+      rightValue = pairSum - leftValue
+    } else if (rightValue < MIN_SEGMENT_PERCENT) {
+      rightValue = Math.ceil(MIN_SEGMENT_PERCENT)
+      leftValue = pairSum - rightValue
+    }
+
+    // Only update the two affected
+    const newValues = [...values]
+    newValues[dragIndex] = leftValue
+    newValues[dragIndex + 1] = rightValue
+    setValues(newValues)
+  }
+
+  // End drag
+  const onMouseUp = () => setDragIndex(null)
+
+  useEffect(() => {
+    if (dragIndex !== null) {
+      window.addEventListener('mousemove', onMouseMove)
+      window.addEventListener('mouseup', onMouseUp)
+    } else {
+      window.removeEventListener('mousemove', onMouseMove)
+      window.removeEventListener('mouseup', onMouseUp)
+    }
+    return () => {
+      window.removeEventListener('mousemove', onMouseMove)
+      window.removeEventListener('mouseup', onMouseUp)
+    }
+    // eslint-disable-next-line
+  }, [dragIndex, values])
+
+  // dnd-kit sort logic
+  function handleDragEnd({ active, over }: DragEndEvent) {
+    if (!over || active.id === over.id) return
+    const oldIndex = itemsData.findIndex(item => item.key === active.id)
+    const newIndex = itemsData.findIndex(item => item.key === over.id)
+    setItemsData(arrayMove(itemsData, oldIndex, newIndex))
+    setValues(arrayMove(values, oldIndex, newIndex))
+  }
+
+  return (
+    <div className={`w-full p-8 ${className}`}>
+      {/* Bar */}
+      <DndContext
+        sensors={sensors}
+        collisionDetection={closestCenter}
```

Comment:
It may be my personal opinion but I like more:
```suggestion
        collisionDetection={pointerWithin}
```

Reply:
While at the beginning I haven't tried because of the following comment regarding the pointer-based only sensors in their documentation, I agree I like it more 👍🏼